staging/src/k8s.io/kubectl/pkg/cmd/attach/attach.go

```diff
@@ -317,3 +318,15 @@ func (o *AttachOptions) GetContainerName(pod *corev1.Pod) (string, error) {
 	}
 	return c.Name, nil
 }
+
+// reattachMessage returns a message to print after attach has completed, or
+// the empty string if no message should be printed.
+func (o *AttachOptions) reattachMessage(containerName string, rawTTY bool) string {
+	if o.Quiet || !o.Stdin || !rawTTY || o.Pod.Spec.RestartPolicy != corev1.RestartPolicyAlways {
+		return ""
+	}
+	if _, path := podcmd.FindContainerByName(o.Pod, containerName); strings.HasPrefix(path, "spec.ephemeralContainers") {
+		return fmt.Sprintf("Session ended, the ephemeral container will not be restarted but may be reattached using '%s %s -c %s -i -t' if it is still running", o.CommandName, o.Pod.Name, containerName)
```

Comment:
In which circumstances ephemeral container can be reattached?, because all of the cases I tried, it failed.

Is it because I exited from ephemeral container by typing`exit`?

Wouldn't be better(or possible) to state that: is reattachment going to work or not instead "may not"?

Reply:
Yes, typing `exit` means that the container is no longer running. If you disconnect while the process is still running (for example, because of a network reset or by killing kubectl) then you can reattach.